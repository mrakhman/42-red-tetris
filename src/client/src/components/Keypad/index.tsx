import { TouchableOpacity, View, Text, ViewStyle } from 'react-native';
import * as React from 'react';


const RoundButton = ({
  size, color, label, style,
}: {
  size: number, color: string, label?: string, style?: ViewStyle,
}): JSX.Element => {
  return (
    <View style={[{ alignItems: 'center', margin: 10 }, style]}>
      <TouchableOpacity
        style={{
          backgroundColor: color,
          borderRadius: size / 2,
          height: size,
          width: size,
          shadowColor: 'rgba(0,0,0, .4)',
          shadowOffset: { height: 1, width: 1 },
          shadowOpacity: 20,
          shadowRadius: 5,
        }} />
      <Text style={{ marginTop: 10 }}>{label}</Text>
    </View>
  );
};

const Keypad = (): JSX.Element => {
  return (
    <View>
      <View style={{ flexDirection: 'row' }}>
        <View style={{ flex: 1 }}>
          <View style={{ flexDirection: 'row', marginTop: 20 }}>
            <RoundButton color="#2dc421" size={50} label="Pause(P)"/>
            <RoundButton color="#2dc421" size={50} label="Sound(S)"/>
            <RoundButton color="#efcc19" size={50} label="Reset(R)"/>
          </View>
          <View style={{ alignItems: 'center' }}>
            <RoundButton color="#5a65f1" size={160} label="Drop(Space)" style={{ marginTop: 20 }}/>
          </View>
        </View>
        <View style={{ alignItems: 'center', flex: 1, justifyContent: 'center' }}>
          <RoundButton color="#5a65f1" size={70} style={{ margin: 0 }} />
          <View style={{ flexDirection: 'row' }}>
            <RoundButton color="#5a65f1" size={70} style={{ margin: 0, marginRight: 60}} />
            <RoundButton color="#5a65f1" size={70} style={{ margin: 0 }} />
          </View>
          <RoundButton color="#5a65f1" size={70} style={{ margin: 0 }} />
        </View>
      </View>
    </View>
  );
};

export default Keypad;
