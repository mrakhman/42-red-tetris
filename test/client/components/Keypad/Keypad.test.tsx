import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Keypad from '/client/components/Keypad';

describe('Snapshots', () => {
  it('should render Keypad for Login and Ranking screens', () => {
    const props = {
      isPause: undefined,
      setIsPause: undefined,
      opponentsNumber: 0,
      isLeader: undefined,
      gameStarted: undefined,
      gameover: undefined,
      isSoloMode: undefined,
      speedMode: undefined,
    };

    const wrapper = shallow(<Keypad {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Keypad for Multiplayer Playground screen: isLeader, not gameStarted, isPause', () => {
    const props = {
      gameover: false,
      gameStarted: false,
      isLeader: true,
      isPause: true,
      isSoloMode: false,
      opponentsNumber: 2,
      setIsPause: jest.fn(),
      speedMode: false,
    };

    const wrapper = shallow(<Keypad {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Keypad for Multiplayer Playground screen: not isLeader, not gameStarted, isPause', () => {
    const props = {
      gameover: false,
      gameStarted: false,
      isLeader: false,
      isPause: true,
      isSoloMode: false,
      opponentsNumber: 2,
      setIsPause: jest.fn(),
      speedMode: false,
    };

    const wrapper = shallow(<Keypad {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Keypad for Multiplayer Playground screen: isLeader, gameStarted, not isPause', () => {
    const props = {
      gameover: false,
      gameStarted: true,
      isLeader: true,
      isPause: false,
      isSoloMode: false,
      opponentsNumber: 2,
      setIsPause: jest.fn(),
      speedMode: false,
    };

    const wrapper = shallow(<Keypad {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Keypad for Multiplayer Playground screen: not isLeader, gameStarted, not isPause', () => {
    const props = {
      gameover: false,
      gameStarted: true,
      isLeader: false,
      isPause: false,
      isSoloMode: false,
      opponentsNumber: 2,
      setIsPause: jest.fn(),
      speedMode: false,
    };

    const wrapper = shallow(<Keypad {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Keypad for Multiplayer Playground screen: not isLeader, gameStarted, isPause, gameover', () => {
    const props = {
      gameover: true,
      gameStarted: true,
      isLeader: false,
      isPause: true,
      isSoloMode: false,
      opponentsNumber: 2,
      setIsPause: jest.fn(),
      speedMode: false,
    };

    const wrapper = shallow(<Keypad {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Keypad for Solo Playground screen: not gameStarted, isPause', () => {
    const props = {
      gameover: false,
      gameStarted: false,
      isLeader: true,
      isPause: true,
      isSoloMode: true,
      opponentsNumber: 0,
      setIsPause: jest.fn(),
      speedMode: false,
    };

    const wrapper = shallow(<Keypad {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Keypad for Solo Playground screen: gameStarted, not isPause', () => {
    const props = {
      gameover: false,
      gameStarted: true,
      isLeader: true,
      isPause: false,
      isSoloMode: true,
      opponentsNumber: 0,
      setIsPause: jest.fn(),
      speedMode: false,
    };

    const wrapper = shallow(<Keypad {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Keypad for Solo Playground screen: gameStarted, isPause', () => {
    const props = {
      gameover: false,
      gameStarted: true,
      isLeader: true,
      isPause: true,
      isSoloMode: true,
      opponentsNumber: 0,
      setIsPause: jest.fn(),
      speedMode: false,
    };

    const wrapper = shallow(<Keypad {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
