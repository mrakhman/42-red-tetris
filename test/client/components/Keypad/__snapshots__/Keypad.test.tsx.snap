// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should render Keypad for Login and Ranking screens 1`] = `
<View
  style={
    Object {
      "width": 400,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
        }
      }
    >
      <RoundButton
        color="#c0c0c0"
        disabled={true}
        label="Reset"
        onPress={[Function]}
        size={50}
      />
      <RoundButton
        color="white"
        label="Exit"
        onPress={[Function]}
        size={50}
        style={
          Object {
            "margin": 0,
          }
        }
        text="â•³"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <RoundButton
          color="#c0c0c0"
          disabled={true}
          label="Drop(Space)"
          onPress={[Function]}
          size={160}
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        <RoundButton
          color="#c0c0c0"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <RoundButton
            color="#c0c0c0"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
                "marginRight": 60,
              }
            }
          />
          <RoundButton
            color="#c0c0c0"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
              }
            }
          />
        </View>
        <RoundButton
          color="#c0c0c0"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <Sound
    autoLoad={true}
    loop={true}
    onBufferChange={[Function]}
    onError={[Function]}
    onFinishedPlaying={[Function]}
    onLoad={[Function]}
    onLoading={[Function]}
    onPause={[Function]}
    onPlaying={[Function]}
    onResume={[Function]}
    onStop={[Function]}
    playStatus="STOPPED"
    playbackRate={1}
    url="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-10.mp3"
    volume={100}
  />
</View>
`;

exports[`Snapshots should render Keypad for Multiplayer Playground screen: isLeader, gameStarted, not isPause 1`] = `
<View
  style={
    Object {
      "width": 400,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
        }
      }
    >
      <RoundButton
        color="#efcc19"
        disabled={false}
        label="Pause"
        onPress={[Function]}
        size={50}
        text="||"
      />
      <RoundButton
        color="white"
        disabled={false}
        label="Super speed"
        onPress={[Function]}
        size={50}
        text="off"
      />
      <RoundButton
        color="#efcc19"
        disabled={false}
        label="Reset"
        onPress={[Function]}
        size={50}
      />
      <RoundButton
        color="white"
        label="Sound"
        onPress={[Function]}
        size={50}
        text="ðŸ¤«"
      />
      <RoundButton
        color="white"
        label="Exit"
        onPress={[Function]}
        size={50}
        style={
          Object {
            "margin": 0,
          }
        }
        text="â•³"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={false}
          label="Drop(Space)"
          onPress={[Function]}
          size={160}
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={false}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <RoundButton
            color="#5a65f1"
            disabled={false}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
                "marginRight": 60,
              }
            }
          />
          <RoundButton
            color="#5a65f1"
            disabled={false}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
              }
            }
          />
        </View>
        <RoundButton
          color="#5a65f1"
          disabled={false}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <Sound
    autoLoad={true}
    loop={true}
    onBufferChange={[Function]}
    onError={[Function]}
    onFinishedPlaying={[Function]}
    onLoad={[Function]}
    onLoading={[Function]}
    onPause={[Function]}
    onPlaying={[Function]}
    onResume={[Function]}
    onStop={[Function]}
    playStatus="STOPPED"
    playbackRate={1}
    url="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-10.mp3"
    volume={100}
  />
</View>
`;

exports[`Snapshots should render Keypad for Multiplayer Playground screen: isLeader, not gameStarted, isPause 1`] = `
<View
  style={
    Object {
      "width": 400,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
        }
      }
    >
      <RoundButton
        color="#2dc421"
        disabled={false}
        label="Start"
        onPress={[Function]}
        size={50}
        text="â–¶"
      />
      <RoundButton
        color="white"
        disabled={false}
        label="Super speed"
        onPress={[Function]}
        size={50}
        text="off"
      />
      <RoundButton
        color="#efcc19"
        disabled={false}
        label="Reset"
        onPress={[Function]}
        size={50}
      />
      <RoundButton
        color="white"
        label="Sound"
        onPress={[Function]}
        size={50}
        text="ðŸ¤«"
      />
      <RoundButton
        color="white"
        label="Exit"
        onPress={[Function]}
        size={50}
        style={
          Object {
            "margin": 0,
          }
        }
        text="â•³"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={true}
          label="Drop(Space)"
          onPress={[Function]}
          size={160}
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <RoundButton
            color="#5a65f1"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
                "marginRight": 60,
              }
            }
          />
          <RoundButton
            color="#5a65f1"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
              }
            }
          />
        </View>
        <RoundButton
          color="#5a65f1"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <Sound
    autoLoad={true}
    loop={true}
    onBufferChange={[Function]}
    onError={[Function]}
    onFinishedPlaying={[Function]}
    onLoad={[Function]}
    onLoading={[Function]}
    onPause={[Function]}
    onPlaying={[Function]}
    onResume={[Function]}
    onStop={[Function]}
    playStatus="STOPPED"
    playbackRate={1}
    url="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-10.mp3"
    volume={100}
  />
</View>
`;

exports[`Snapshots should render Keypad for Multiplayer Playground screen: not isLeader, gameStarted, isPause, gameover 1`] = `
<View
  style={
    Object {
      "width": 400,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
        }
      }
    >
      <RoundButton
        color="#c0c0c0"
        disabled={true}
        label="Super speed"
        onPress={[Function]}
        size={50}
        text="off"
      />
      <RoundButton
        color="#c0c0c0"
        disabled={true}
        label="Reset"
        onPress={[Function]}
        size={50}
      />
      <RoundButton
        color="white"
        label="Sound"
        onPress={[Function]}
        size={50}
        text="ðŸ¤«"
      />
      <RoundButton
        color="white"
        label="Exit"
        onPress={[Function]}
        size={50}
        style={
          Object {
            "margin": 0,
          }
        }
        text="â•³"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <RoundButton
          color="#c0c0c0"
          disabled={true}
          label="Drop(Space)"
          onPress={[Function]}
          size={160}
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        <RoundButton
          color="#c0c0c0"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <RoundButton
            color="#c0c0c0"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
                "marginRight": 60,
              }
            }
          />
          <RoundButton
            color="#c0c0c0"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
              }
            }
          />
        </View>
        <RoundButton
          color="#c0c0c0"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <Sound
    autoLoad={true}
    loop={true}
    onBufferChange={[Function]}
    onError={[Function]}
    onFinishedPlaying={[Function]}
    onLoad={[Function]}
    onLoading={[Function]}
    onPause={[Function]}
    onPlaying={[Function]}
    onResume={[Function]}
    onStop={[Function]}
    playStatus="STOPPED"
    playbackRate={1}
    url="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-10.mp3"
    volume={100}
  />
</View>
`;

exports[`Snapshots should render Keypad for Multiplayer Playground screen: not isLeader, gameStarted, not isPause 1`] = `
<View
  style={
    Object {
      "width": 400,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
        }
      }
    >
      <RoundButton
        color="white"
        disabled={false}
        label="Super speed"
        onPress={[Function]}
        size={50}
        text="off"
      />
      <RoundButton
        color="#efcc19"
        disabled={false}
        label="Reset"
        onPress={[Function]}
        size={50}
      />
      <RoundButton
        color="white"
        label="Sound"
        onPress={[Function]}
        size={50}
        text="ðŸ¤«"
      />
      <RoundButton
        color="white"
        label="Exit"
        onPress={[Function]}
        size={50}
        style={
          Object {
            "margin": 0,
          }
        }
        text="â•³"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={false}
          label="Drop(Space)"
          onPress={[Function]}
          size={160}
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={false}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <RoundButton
            color="#5a65f1"
            disabled={false}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
                "marginRight": 60,
              }
            }
          />
          <RoundButton
            color="#5a65f1"
            disabled={false}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
              }
            }
          />
        </View>
        <RoundButton
          color="#5a65f1"
          disabled={false}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <Sound
    autoLoad={true}
    loop={true}
    onBufferChange={[Function]}
    onError={[Function]}
    onFinishedPlaying={[Function]}
    onLoad={[Function]}
    onLoading={[Function]}
    onPause={[Function]}
    onPlaying={[Function]}
    onResume={[Function]}
    onStop={[Function]}
    playStatus="STOPPED"
    playbackRate={1}
    url="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-10.mp3"
    volume={100}
  />
</View>
`;

exports[`Snapshots should render Keypad for Multiplayer Playground screen: not isLeader, not gameStarted, isPause 1`] = `
<View
  style={
    Object {
      "width": 400,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
        }
      }
    >
      <RoundButton
        color="white"
        disabled={false}
        label="Super speed"
        onPress={[Function]}
        size={50}
        text="off"
      />
      <RoundButton
        color="#efcc19"
        disabled={false}
        label="Reset"
        onPress={[Function]}
        size={50}
      />
      <RoundButton
        color="white"
        label="Sound"
        onPress={[Function]}
        size={50}
        text="ðŸ¤«"
      />
      <RoundButton
        color="white"
        label="Exit"
        onPress={[Function]}
        size={50}
        style={
          Object {
            "margin": 0,
          }
        }
        text="â•³"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={true}
          label="Drop(Space)"
          onPress={[Function]}
          size={160}
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <RoundButton
            color="#5a65f1"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
                "marginRight": 60,
              }
            }
          />
          <RoundButton
            color="#5a65f1"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
              }
            }
          />
        </View>
        <RoundButton
          color="#5a65f1"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <Sound
    autoLoad={true}
    loop={true}
    onBufferChange={[Function]}
    onError={[Function]}
    onFinishedPlaying={[Function]}
    onLoad={[Function]}
    onLoading={[Function]}
    onPause={[Function]}
    onPlaying={[Function]}
    onResume={[Function]}
    onStop={[Function]}
    playStatus="STOPPED"
    playbackRate={1}
    url="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-10.mp3"
    volume={100}
  />
</View>
`;

exports[`Snapshots should render Keypad for Solo Playground screen: gameStarted, isPause 1`] = `
<View
  style={
    Object {
      "width": 400,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
        }
      }
    >
      <RoundButton
        color="#efcc19"
        disabled={false}
        label="Play"
        onPress={[Function]}
        size={50}
        text="â–¶"
      />
      <RoundButton
        color="white"
        disabled={false}
        label="Super speed"
        onPress={[Function]}
        size={50}
        text="off"
      />
      <RoundButton
        color="#efcc19"
        disabled={false}
        label="Reset"
        onPress={[Function]}
        size={50}
      />
      <RoundButton
        color="white"
        label="Sound"
        onPress={[Function]}
        size={50}
        text="ðŸ¤«"
      />
      <RoundButton
        color="white"
        label="Exit"
        onPress={[Function]}
        size={50}
        style={
          Object {
            "margin": 0,
          }
        }
        text="â•³"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={true}
          label="Drop(Space)"
          onPress={[Function]}
          size={160}
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <RoundButton
            color="#5a65f1"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
                "marginRight": 60,
              }
            }
          />
          <RoundButton
            color="#5a65f1"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
              }
            }
          />
        </View>
        <RoundButton
          color="#5a65f1"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <Sound
    autoLoad={true}
    loop={true}
    onBufferChange={[Function]}
    onError={[Function]}
    onFinishedPlaying={[Function]}
    onLoad={[Function]}
    onLoading={[Function]}
    onPause={[Function]}
    onPlaying={[Function]}
    onResume={[Function]}
    onStop={[Function]}
    playStatus="STOPPED"
    playbackRate={1}
    url="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-10.mp3"
    volume={100}
  />
</View>
`;

exports[`Snapshots should render Keypad for Solo Playground screen: gameStarted, not isPause 1`] = `
<View
  style={
    Object {
      "width": 400,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
        }
      }
    >
      <RoundButton
        color="#efcc19"
        disabled={false}
        label="Pause"
        onPress={[Function]}
        size={50}
        text="||"
      />
      <RoundButton
        color="white"
        disabled={false}
        label="Super speed"
        onPress={[Function]}
        size={50}
        text="off"
      />
      <RoundButton
        color="#efcc19"
        disabled={false}
        label="Reset"
        onPress={[Function]}
        size={50}
      />
      <RoundButton
        color="white"
        label="Sound"
        onPress={[Function]}
        size={50}
        text="ðŸ¤«"
      />
      <RoundButton
        color="white"
        label="Exit"
        onPress={[Function]}
        size={50}
        style={
          Object {
            "margin": 0,
          }
        }
        text="â•³"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={false}
          label="Drop(Space)"
          onPress={[Function]}
          size={160}
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={false}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <RoundButton
            color="#5a65f1"
            disabled={false}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
                "marginRight": 60,
              }
            }
          />
          <RoundButton
            color="#5a65f1"
            disabled={false}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
              }
            }
          />
        </View>
        <RoundButton
          color="#5a65f1"
          disabled={false}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <Sound
    autoLoad={true}
    loop={true}
    onBufferChange={[Function]}
    onError={[Function]}
    onFinishedPlaying={[Function]}
    onLoad={[Function]}
    onLoading={[Function]}
    onPause={[Function]}
    onPlaying={[Function]}
    onResume={[Function]}
    onStop={[Function]}
    playStatus="STOPPED"
    playbackRate={1}
    url="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-10.mp3"
    volume={100}
  />
</View>
`;

exports[`Snapshots should render Keypad for Solo Playground screen: not gameStarted, isPause 1`] = `
<View
  style={
    Object {
      "width": 400,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
        }
      }
    >
      <RoundButton
        color="#2dc421"
        disabled={false}
        label="Start"
        onPress={[Function]}
        size={50}
        text="â–¶"
      />
      <RoundButton
        color="white"
        disabled={false}
        label="Super speed"
        onPress={[Function]}
        size={50}
        text="off"
      />
      <RoundButton
        color="#efcc19"
        disabled={false}
        label="Reset"
        onPress={[Function]}
        size={50}
      />
      <RoundButton
        color="white"
        label="Sound"
        onPress={[Function]}
        size={50}
        text="ðŸ¤«"
      />
      <RoundButton
        color="white"
        label="Exit"
        onPress={[Function]}
        size={50}
        style={
          Object {
            "margin": 0,
          }
        }
        text="â•³"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={true}
          label="Drop(Space)"
          onPress={[Function]}
          size={160}
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        <RoundButton
          color="#5a65f1"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <RoundButton
            color="#5a65f1"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
                "marginRight": 60,
              }
            }
          />
          <RoundButton
            color="#5a65f1"
            disabled={true}
            onPress={[Function]}
            size={70}
            style={
              Object {
                "margin": 0,
              }
            }
          />
        </View>
        <RoundButton
          color="#5a65f1"
          disabled={true}
          onPress={[Function]}
          size={70}
          style={
            Object {
              "margin": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <Sound
    autoLoad={true}
    loop={true}
    onBufferChange={[Function]}
    onError={[Function]}
    onFinishedPlaying={[Function]}
    onLoad={[Function]}
    onLoading={[Function]}
    onPause={[Function]}
    onPlaying={[Function]}
    onResume={[Function]}
    onStop={[Function]}
    playStatus="STOPPED"
    playbackRate={1}
    url="https://www.soundhelix.com/examples/mp3/SoundHelix-Song-10.mp3"
    volume={100}
  />
</View>
`;
